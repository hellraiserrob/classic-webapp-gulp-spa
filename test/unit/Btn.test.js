import { shallow } from '@vue/test-utils';
import MyButton from '../../src/scripts/components/Btn.vue';

describe('Btn.vue', () => {
  
  let wrapper;
  const label = 'new message';
  const action = jest.fn();

  beforeEach(() => {
    wrapper = shallow(MyButton, {
      slots: {
        default: label,
      },
      propsData: {
        action,
      }
    });
  });

  it('renders label when passed in default slot', () => {
    expect(wrapper.text()).toMatch(label);
  });

  it('action props should be called on click', () => {
    wrapper.trigger('click');
    expect(action).toHaveBeenCalled();
  });
  
  it('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

});
