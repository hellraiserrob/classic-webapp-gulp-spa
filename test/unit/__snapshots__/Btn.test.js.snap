// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn.vue Should match snapshot 1`] = `<button class="btn">new message</button>`;
